airbyte-integrations/connectors/destination-s3-glue/src/main/java/io/airbyte/integrations/destination/s3_glue/GlueOperations.java

```diff
@@ -109,29 +112,39 @@ public void deleteTable(String databaseName, String tableName) {
 
   }
 
-  private Collection<Column> transformSchema(JsonNode jsonSchema) {
+  private Collection<Column> transformSchema(JsonNode jsonSchema, MetastoreFormatConfig metastoreFormatConfig) {
     if (jsonSchema.has("properties")) {
       Map<String, JsonNode> properties = objectMapper.convertValue(jsonSchema.get("properties"), new TypeReference<>() {});
       return properties.entrySet().stream()
-          .map(es -> new Column().withName(es.getKey()).withType(transformSchemaRecursive(es.getValue())))
+          .map(es -> new Column().withName(es.getKey()).withType(transformSchemaRecursive(es.getValue(), metastoreFormatConfig)))
           .collect(Collectors.toSet());
     } else {
       return Collections.emptySet();
     }
   }
 
-  private String transformSchemaRecursive(JsonNode jsonNode) {
+  private String transformSchemaRecursive(JsonNode jsonNode, MetastoreFormatConfig metastoreFormatConfig) {
     String type = filterTypes(jsonNode.get("type")).iterator().next();
     return switch (type) {
       // TODO(itaseski) support date-time and timestamp airbyte types
       case "string" -> "string";
       case "number" -> {
         if (jsonNode.has("airbyte_type") && jsonNode.get("airbyte_type").asText().equals("integer")) {
           yield "int";
+        } else {
+          if (metastoreFormatConfig.getNumericType() == NumericType.FLOAT) {
+            yield "float";
+          } else {
+            if (metastoreFormatConfig.getNumericType() == NumericType.DECIMAL) {
+              yield String.format("decimal(38,%s)", metastoreFormatConfig.getDecimalScale());
+            } else {
+              // TODO(quazi-h) We have set the default parameter to 2 in the spec for this parameter
+              // Default to use decimal as it is a more precise type and allows for large values
+              // Set the default scale 38 to allow for the widest range of values
+              yield "decimal(38,2)";
```

Comment:
Can I remove this default since we should be using the default of "2" set in the connector spec?